qiskit/transpiler/target.py

```diff
@@ -1311,151 +1314,218 @@ def from_configuration(
+                            # Associated qubit set might be defined by the gate config here
+                            logger.info(
+                                "Gate properties of instruction %s are not provided for every qubits. "
+                                "This gate is ideal for some qubits and the rest is with finite error. "
+                                "Created backend target may confuse error-aware circuit optimization.",
+                                name,
+                            )
+                except BackendPropertyError:
+                    # This gate doesn't report any property
+                    continue
+
+            # Measure instruction property is stored in qubit property
+            prop_name_map["measure"] = {}
+            for qubit_idx in range(num_qubits):
+                qubit_prop = backend_properties.qubit_property(qubit_idx)
+                duration = (
+                    qubit_prop["readout_length"][0] if "readout_length" in qubit_prop else None
+                )
+                error = qubit_prop["readout_error"][0] if "readout_error" in qubit_prop else None
+                prop_name_map["measure"][(qubit_idx,)] = InstructionProperties(
+                    duration=duration, error=error
+                )
+
+        if num_qubits is not None:
+            for op in required_operations:
+                # Map required ops to each operational qubit
+                if prop_name_map[op] is None:
+                    prop_name_map[op] = {
+                        (q,): None for q in range(num_qubits) if q not in faulty_qubits
+                    }
+
+        if inst_map is not None and coupling_map is not None:
+            for name in inst_map.instructions:
+                for qubits in inst_map.qubits_with_instruction(name):
+                    if not isinstance(qubits, tuple):
+                        qubits = (qubits,)
+
+                    if name not in prop_name_map or qubits not in prop_name_map[name]:
+                        logger.info(
+                            "Gate calibration for instruction %s on qubits %s is found "
+                            "in the PulseDefaults payload. However, this entry is not defined in "
+                            "the gate mapping of Target. This calibration is ignored.",
+                            name,
+                            qubits,
                         )
-                target.add_instruction(name_mapping[gate], properties=gate_properties, name=gate)
-            for gate in global_ideal_variable_width_gates:
-                target.add_instruction(name_mapping[gate], name=gate)
+                        continue
+
+                    if (name, qubits) in faulty_ops:
+                        continue
+
+                    entry = inst_map._get_calibration_entry(name, qubits)
+
+                    try:
+                        prop_name_map[name][qubits].calibration = entry
```

Comment:
You also need to fill gate duration based on the calibrated schedule `.duration` when the `InstructionProperties.duration` for this entry is missing. The docstring says duration can also come from the calibrated schedule when other two sources don't provide any information.

Reply:
I was not sure if the docstrings should be changed or the logic itself should be changed. That's why I request your guidance :)
Applied suggestion :)